### DISCUSSION TITLE
UI proposal

### DISCUSSION THREAD

**Discussion by Djioup** [T-Troll/alienfx-tools#96] [Ideas]
I'm a UI apprentice, I'd be very happy to give you some ideas for the interface of your application.

If you're ever interested, I've made a little draft that you can test quickly : 
- [Test 1](https://xd.adobe.com/view/0479cabf-472e-468f-b8d3-484c3d5cf2ff-512a/?fullscreen) (It's just a draft, there is only one interactive button)
- [Test 2](https://xd.adobe.com/view/8ebb43a8-c0b1-434c-8f73-5b759327881a-dbe8/?fullscreen)

PS : I got some images from the internet for the draft, I didn't make them manually.

**Comment by T-Troll** [comment]
Thank you. Feel free to share any ideas, it's make sense in any case!

But let me share some points of my vision.

I don't like to make another Gb-size monster like Dell's one. So i try to keep it as simple and tiny as possible, that's why i use sole Windows controls. Maybe later it will be good to support skins as an option, but as an option only.

But i have another issue. I don't like some current UI choices from the UX position - it's so many features not explained well and not easy to understand and use. In fact, the current app is a monster already, far beyond the initial design concept. And here are a lot of points to re-work, simplify and explain.

However, i position this tool as a more professional one (there are still some ways to shoot yourself into the leg now).

So, from priority list, UX first - even if it's requiring full UI rebuild. Visual pleasure is second.

PS: At the beginning, i just want to create a lights control CLI for my gear. Now it's come so far...

**Comment by Djioup** [comment]
Okay, got it. I'll try to come back with other proposals much simpler and lighter. You want to use the Windows controls necessarily if I understand correctly?

Otherwise, I proposed to separate the tool in 2 parts: light on one side and fan on the other because finally, they are the two main axis of the application and I think it would make the understanding much easier. What do you think about it ?

**Reply by T-Troll** [reply]
No. The target is to keep application slim, You see, now ALL application size is <450Kb. Your background picture is already bigger. ^_^

Yes, this is reasonable. In fact, i add fans in FX GUI for just not to keep 2 apps in tray. ^_^
Also, "Fans" is not exactly right name. It's "Fan and power control", in fact. I plan to add even more power control there.

**Comment by Djioup** [comment]
Okay, I see. I'll go like this. I'll be back in the next few days.

**Reply by T-Troll** [reply]
You are welcome!

**Comment by Djioup** [comment]
I test an approach, you tell me what you think.
![Plan de travail 1](https://user-images.githubusercontent.com/78264593/145712493-5e7b1897-c473-47f8-be8c-64298ba9e687.png)



Everything is done in vector to be as light as possible (30 to 40 Ko for these examples). This would be the home screen. By clicking on the icons, the tabs open with more details. I'm not convinced by the Light FX UI and by the colors as a whole and I have other ideas for the layout of the icons, it's mostly the style of the background that I question here. Do you like this purified approach ? Don't be afraid to tell me if you don't like it.

**Reply by T-Troll** [reply]
Minimalistic. So it's a good direction.

But there are some issues - f.e. full-vector UI require a lot of CPU to draw (i know it because i work with maps before).

So... Good idea about vector ELEMENTS, it can really save the day.

Can i ask you to think a bit out-of the box? You know functionality modules (i can share connections between them if you wish). Can you try to make a UI tree - what info here, here, here and how it's connected? Art is next, let's start from core UI structure/UX!

**Comment by Djioup** [comment]
OK I get it. I'm going to make the UI tree.

**Reply by T-Troll** [reply]
You can use RUP's use-cases approach.
